### ISSUE TITLE
stereo side-by-side mode broken by commit 89c5590a2ecc3132d34fa5b692d03b96e58ac117

### ISSUE THREAD

**Issue by jiribohac** [BestImageViewer/geeqie#892]
### ISSUE TYPE
* Bug Report

### GEEQIE VERSION
```
Geeqie 1.6+git20210408-a33d8c40 GTK3
```

```
     Config results: 
    -=-=-=-=-=-=-=-=- 
 
Package: 
  Name:          geeqie 
  Version:       1.6+git20210408-a33d8c40 
 
Architecture: 
  UNIX:          yes 
  Win32:         no (native: no) 
 
Flags: 
  Geeqie:         
  DEFS:          -DHAVE_CONFIG_H 
  CPPFLAGS:       
  CFLAGS:         -Werror -Wno-error=deprecated-declarations -Wno-error=sign-compare -Wno-error=return-type 
  CXXFLAGS:      -g -O2 
  Gtk:           -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/libdrm 
  Glib:          -pthread -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include 
  Thread:        -lgthread-2.0 -pthread -lglib-2.0 
  Others:        -ljpeg -ltiff -llcms2 -lexiv2 -lclutter-1.0 -lcogl-path -latk-1.0 -lcogl-pango -lcogl -lgmodule-2.0 -pthread -lwayland-egl -lgbm -ldrm -lEGL -lXrandr -ljson-glib-1.0 -lgio-2.0 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lgdk_pixbuf-2.0 -lcairo-gobject -lcair
 
Localization: 
  NLS support:   yes 
  LINGUAS:        ar be bg ca cs da de el en_GB eo es et eu fi fr hu id it ja ko nb nl pl pt_BR ro ru sk sl sr@latin sr sv th tlh tr uk vi zh_CN zh_TW 
 
Settings: 
  Developer:     no 
  Debug flags:   no 
  Debug log:     yes 
  Deprecated:    no 
 
Support: 
  LCMS:          yes 
  Exiv2:         yes 
  Lirc:          no 
  Clutter:       yes 
  Libchamplain:         yes 
  Libchamplain-gtk:     yes 
  Lua:           yes 
  FFmpegthumbnailer:    yes 
  Pdf:           yes 
  HEIF:          yes 
  WebP:          yes 
  DjVu:          yes 
  J2K:           yes 
```

### OS / DISTRIBUTION
Debian stable
(I backported geeqie from testing because of something else, found this problem, so I compiled geeqie myself to perform a bisect)

### SUMMARY
Stereo side-by-side mode is completely broken; the right side seems to be OK (but sometimes shows the full image, not just the right side); the left side shows something strange, not the left side of the image. The left side does not redraw when moving to the next image. Happens in both full-screen and windowed mode.

### STEPS TO REPRODUCE
- go to Preferences -> Stereo -> Windowed stereo mode; choose "Side by side"
- go to View -> Stereo -> Side by side
- view any jpeg file; it is supposed to consider one half as the stereo left side and one half as the stereo right side; both sides should be shown next to each other, possibly with a dividing black strip between them.
- observe the left side - it shows nonsense; the right side is sometimes ok, sometimes shows the full picture and not just the righ half

I bisected the problem and found 89c5590a2ecc3132d34fa5b692d03b96e58ac117 ("Fix #644: Images fail to render on MacOS") to cause the problem.

(to test on current master, reverting 84d758f0a1aef90b4890538e952d37b00542aef7 first will allow to revert 89c5590a2ecc3132d34fa5b692d03b96e58ac117 cleanly)

**Comment by caclark** [comment]
An immediate work-around is to select `Edit/Preferences/Image/Use GPU Acceleration` or compile for GTK2 `autogen.sh --disable-gtk3`

**Comment by jiribohac** [comment]
Thanks, but none of this works for me; the Acceleration settings is grayed on my machine and I can't use GTK2 because the fix for #651 only works with GTK3. Let me know if I can do anything to help resolve this.

**Comment by xiota** [comment]
@jiribohac To enable GPU Acceleration:

* `Edit/Preferences/Behaviour/Override disable GPU`
* Restart Geeqie
* `Edit/Preferences/Image/Use GPU Acceleration`
* Restart Geeqie
